### ISSUE TITLE
New Install of Autopsy 4.0.0: Ingest Failure on E01?

### ISSUE THREAD

**Issue by MITService** [sleuthkit/autopsy#2108]
New install and have an E01 that I'm trying to process an image. It loads up fine, reports work, etc.  I get an ingest failure when it's parsing, so I'm not getting any keyword search module results. It says the following

Keyword Search:  The index could not be opened or does not exist

This is a fresh install from the web, and shows 7 installed plug-ins. I don't see any updates and don't see what's causing the issue.

I have tryied adding search keyword index list, etc.  I can not get it to search anything.  The E01 opens and shows 4 volumes

Thanks!!

**Comment by karlmortensen** [comment]
Is this a multi-user case? If so, are you storing the results on drive that the remote Solr server is unable to write to (such as a local C:\some\folder)? If this is the case, simply store the results on some network shared drive with a UNC path and this should work e.g. use \\somehostname\foldername\blah

**Comment by MITService** [comment]
It is not a multi user case.  It is on a D drive.  Local HDD.

I will try moving the image to the network and see what happens.

**Comment by karlmortensen** [comment]
If it's a single-user case, placing it on the network won't help, as it should be running a local Solr instance.

**Comment by rcordovano** [comment]
There is a known problem with the version of Solr that Autopsy is using
that causes Solr to mix up its state files for single-user cases. Just to
be sure we are not overlooking a possible solution, did you try closing and
reopening Autopsy? Doing so causes Solr to sort things out, if that is the
problem you are experiencing.

**Comment by MITService** [comment]
I have closed it and rebooted the server.

**Comment by rcordovano** [comment]
Here is what a healthy situation looks like for a *single-user* case with a
text index:

- The XML in the case metadata file (.aut extension) in the case folder
 should have a TextIndexName element with a text index name as its content.
- The ModuleOutput\keywordsearch\data\index subdirectory of the case folder
should contain a bunch of files. These constitute the text index (Solr
core) for the case.
- There should be a subdirectory of the solr\solr subdirectory of the
Autopsy installation directory with the same name as the text index name
from the case metadata file (e.g., C:\Program
Files\Autopsy-4.0.0\autopsy\solr\solr\<text index name>).  Inside that
directory there should be a file named core.properties (if the problem I
mentioned earlier was in play, the file would have an additional "unloaded"
extension tacked on to it; restarting Autopsy restarts the local Solr
server [running in jetty and appearing as java.exe in Task Manager; a Core
admin page may be reached by browser at http://localhost:23232/solr/#/~cores
when it is running] which then figures out that it needs to strip off the
erroneously added extension).

Assuming all of that is as it should be, the next step would be to look
through the Autopsy logs for the case for clues (or send them to me a
rcordovano@basistech.com). The Autopsy logs for a case are located at in
the test\Log subdirectory of the case folder; you can get there from
Autopsy via Help, Open Log Folder. The traces log has stack traces, so that
is the log I prefer to read rather than the "user-friendly" log files (sans
stack traces).  Additional logs and the "in between cases" Autopsy log can
be found in the var/log subdirectory of the directory identified as
"Userdir" in the Help, About dialog.

**Comment by rcordovano** [comment]
Any progress towards resolution, MITService?

**Comment by mraggi416** [comment]
I am having the exact same issue. I was wondering if anyone has been able to resolve this?

**Comment by rcordovano** [comment]
I am not aware of a resolution, and we recently received a report of
something that sounds similar with Autopsy 4.2.0. Here is where we stand
with that:

"Nanni, thanks for your cooperation in our efforts to solve this problem.
Three of us have combed through the log files - a senior developer, our
Systems Administrator, and myself - and we can see that some other process
is interfering with writes to your Solr index when you first create a case:

612073 [qtp329611835-12] ERROR org.apache.solr.update.SolrIndexWriter  –
Error closing IndexWriter
java.nio.file.FileSystemException:
C:\pen256\1gb\1gb\ModuleOutput\keywordsearch\data\index\_q_Lucene41_0.doc:
Impossibile accedere al file. Il file è utilizzato da un altro processo.

Unfortunately, we have been unable to reproduce the problem. It seems that
it is specific to your environment. One idea we have had is that perhaps an
antivirus program is taking a lock on the _q_Lucene41_0.doc file to examine
it. So if you are running AV software, you might like to test this
hypothesis by temporarily turning the AV off. If we had access to your
machine, we would run SysInternals ProcessExplorer (
https://technet.microsoft.com/en-us/sysinternals/bb545021.aspx) to try to
detect which processes had open handles to this file when the incident
occurs, but we have reached a dead end in what we can do to investigate
this problem from our offices."

If you are able to send me a copy of ~/Users/[your user
name]/AppData/roaming/autopsy/var/log/solr.stdout.log to me, I can verify
whether or not it is the same problem, or you could look yourself. If you
do decide to look into this log file or send me a copy, be sure that you
collect the file while you are in the "The index could not be opened or
does not exist" state. The file is overwritten when you open a new case.

Thanks,

Richard Cordovano
Director of Engineering, Digital Forensics
Basis Technology

**Comment by Sid-Vishez** [comment]
Have same issue. Resolved after I have set a proxy settings to "No proxy".

**Comment by deralexxx** [comment]
indeed, setting proxy to no proxy helped me as well.

**Comment by deralexxx** [comment]
any chance to get that fixed?

**Comment by bcarrier** [comment]
So, to summarize the issue:
- Autopsy by default has a setting for "Use System Proxy Settings" (under the General Tab in the Options panel).
- I'm assuming that your OS has a proxy configured, yes?
- When you start Autopsy, Keyword search complains.
- You changed the proxy setting to "No Proxy" and then Keyword search is happy

Is this correct?

If so, this may make sense. The proxy setting is primarily used to get notifications of new updates (and other modules that may need to reach out to a site).   Autopsy tries to connect to the local Solr server by referring to it as "localhost". 

I would have hoped that the OS would not send localhost traffic to a proxy, but maybe it is.

**Comment by deralexxx** [comment]
Yes that is correct, but localhost and 127.0.0.1 are on the exception list for proxy (of course)

Shall I take a traffic capture to see what is happening?

**Comment by Trusquin** [comment]
I installed Autopsy 4.8.0 on a Mint 19 XFCE in a Virtual Box VM and had the "ERROR org.apache.solr.update.SolrIndexWriter". I carefully read this forum entry and applied all the recommendations and still had the same error after rebooting the whole system. I then installed Sol'r on the system, started the service and Autopsy is now able to work properly. I don't know if I succeded by luck or if it's logic.
